tasks/core/apps.yml

```diff
@@ -38,143 +38,101 @@
         ]
       }}
 
-- name: Install external apps
-  block:
-    - name: Find full nextcloud version
-      command: ./occ status  --output=json
-      args:
-        chdir: "{{ nextcloud_installation_dir }}"
-      become: true
-      become_user: "{{ nextcloud_file_owner }}"
-      register: _result
-      changed_when: false
-
-    - name: Remove non-json text from command output
-      set_fact:
-        nextcloud_full_version:
-          "{{ (_result.stdout | from_json)['versionstring'] }}"
-
-    - name: Create app json file url
-      set_fact:
-        nextcloud_app_store_json_url: >-
-          https://apps.nextcloud.com/api/v1/platform/{{
-            nextcloud_full_version
-          }}/apps.json
-      changed_when: false
-
-    - name: Download the app json file
-      get_url:
-        url: "{{ nextcloud_app_store_json_url }}"
-        dest: /tmp
-      register: result
-      until: result is success
+# Remove all apps from the list of external apps which
+# - is not in the configured list of apps
+# - is in the configured list of apps but have their state set to "absent"
+- name: Remove external apps
+  command: php occ app:remove "{{ item }}"
+  with_items: "{{ (nextcloud_installed_apps.enabled |
+    combine(nextcloud_installed_apps.disabled)) }}"
+  become_user: "{{ nextcloud_file_owner }}"
```

Comment:
This `become_user` and the others should have an accompanying `become: true`. Then the tests will be OK.